xds/internal/balancer/edsbalancer/config.go

```diff
@@ -42,6 +42,10 @@ type EDSConfig struct {
 	// will be disabled.  If set to the empty string, load reporting will
 	// be sent to the same server that we obtained CDS data from.
 	LrsLoadReportingServerName *string
+	// If circuit breaking is enabled.
+	CircuitBreaking bool
+	// Max requests when circuit breaking.
+	MaxRequests uint32
```

Comment:
With a second thought, we might not need this field.

The reason we need this is to update the upper bound in the counter.
There are two approaches:

1. keep this field, and store it inside the eds.dropPicker. So the global counter doesn't keep the upper bound, but each picker keeps it. And we update the picker when this value changes.
2. remove this field (so there's no change to the EDSConfig). Inside the CDS balancer, when it receives the ClusterUpdate, it updates the upper of the corresponding counter.

Now I'm a bit in favor of 2. Maybe give it a try. (Keep them as separate commits though, in case we regret later.)

Reply:
I implemented 1 in this branch and pushed it. I'll implement 2 in a separate branch.

Reply:
https://github.com/GarrettGutierrez1/grpc-go/tree/xds-circuit-breaking-no-edsconfig-change

This branch implements 2. I do think it is simpler (less files changed).

Reply:
I switched it up, this branch now implements 2, and the following link implements 1.

https://github.com/GarrettGutierrez1/grpc-go/tree/xds-circuit-breaking-moved-max-requests-tracking